sphinx/ext/apidoc.py

```diff
@@ -194,16 +194,15 @@ def shall_skip(module, opts, excludes=[]):
     if not opts.implicit_namespaces and not path.exists(module):
         return True
 
-    # skip it if there is nothing (or just \n or \r\n) in the file
-    if path.exists(module) and path.getsize(module) <= 2:
-        if os.path.basename(module) == '__init__.py':
-            # We only want to skip packages if they do not contain any
-            # .py files other than __init__.py.
-            basemodule = path.dirname(module)
-            for module in glob.glob(path.join(basemodule, '*.py')):
-                if not is_excluded(path.join(basemodule, module), excludes):
-                    return True
-        else:
+    # Are we a package (here defined as __init__.py, not the folder in itself)
+    if os.path.basename(module) == INITPY:
+        # Yes, check if we have any non-excluded modules at all here
+        basemodule = path.dirname(module)
+        for module in glob.glob(path.join(basemodule, '*.py')):
+            if not is_excluded(path.join(basemodule, module), excludes):
+                # There's a non-excluded module here, we won't skip
+                all_skipped = False
+        if all_skipped:
```

Comment:
It seems this variable does not initialized. Did you test a case all-skipped?

Reply:
Oh, all_skipped should indeed be initialized. However, the case never occurs, unless __init__.py because we're looking for .py files in the same folder as __init__.py - and __init__.py will be there (unless it is matched by the excludes).

Since this commit removes the requirement of skipping modules with an empty __init__.py, I think this entire loop becomes unnecessary - we can simply check if the filename is __init__.py and return false. When we start doing that, we can also remove all calls to shall_skip with an __init__.py as argument, and the code will be vastly simplified.

I'll try to take a deeper look at this this weekend.

Reply:
I got the error with following testcase:
```
@pytest.mark.apidoc(coderoot='test-apidoc-toc/mypackage', excludes=['something/__init__.py'])
def test(apidoc):
    pass
```
```
        # Are we a package (here defined as __init__.py, not the folder in itself)
        if os.path.basename(module) == INITPY:
            # Yes, check if we have any non-excluded modules at all here
            basemodule = path.dirname(module)
            for module in glob.glob(path.join(basemodule, '*.py')):
                if not is_excluded(path.join(basemodule, module), excludes):
                    # There's a non-excluded module here, we won't skip
                    all_skipped = False
>           if all_skipped:
E           UnboundLocalError: local variable 'all_skipped' referenced before assignment
```

Reply:
@zufallsgenerator I'd like to release 1.7.2 today. So I'd like to fix the UnboundLocalError and merge it (I added the fix for it on #4766).
I think simplifier is a refactoring. So it is okay after the release.
What do you think?